pkg/controller/seed-controller-manager/addon/addon_controller.go

```diff
@@ -260,12 +276,12 @@ func (r *Reconciler) reconcile(ctx context.Context, log *zap.SugaredLogger, addo
 	}
 
 	// Reconciling
-	if err := r.ensureIsInstalled(ctx, log, addon, cluster); err != nil {
-		return nil, fmt.Errorf("failed to deploy the addon manifests into the cluster: %v", err)
-	}
 	if err := r.ensureFinalizerIsSet(ctx, addon); err != nil {
 		return nil, fmt.Errorf("failed to ensure that the cleanup finalizer exists on the addon: %v", err)
 	}
+	if err := r.ensureIsInstalled(ctx, log, addon, cluster); err != nil {
```

Comment:
I think this was done the other way around and potentially could have been a culprit for leaving residue behind if someone deleted an addon in the middle of an installation before the finalizers had a chance to be added.